src/lib/option.ts

```diff
@@ -76,118 +73,106 @@ export class Opt<T = unknown> extends Variable<T> {
     }
 
     /**
-     * Initializes this option by attempting to read its value from a cache file.
-     * If found, sets the current value. Also sets up a subscription to write updates back.
+     * Initializes this option based on the provided configuration, if available.
      *
-     * @param cacheFile - The path to the cache file.
+     * @param config - The configuration.
      */
-    public init(cacheFile: string): void {
-        const rawData = readFile(cacheFile);
-
-        let cacheData: Record<string, unknown> = {};
-
-        if (rawData && rawData.trim() !== '') {
-            try {
-                cacheData = JSON.parse(rawData) as Record<string, unknown>;
-            } catch (error) {
-                errorHandler(error);
-            }
-        }
-
-        const cachedVariable = this._findKey(cacheData, this._id.split('.'));
+    public init(config: Record<string, unknown>): void {
+        const value = _findVal(config, this._id.split('.'));
 
-        if (cachedVariable !== undefined) {
-            this.set(cachedVariable as T);
+        if (value !== undefined) {
+            this.set(value as T, { writeDisk: false });
         }
-
-        this.subscribe((newVal) => {
-            const reRaw = readFile(cacheFile);
-            let currentCache: Record<string, unknown> = {};
-            if (reRaw && reRaw.trim() !== '') {
-                try {
-                    currentCache = JSON.parse(reRaw) as Record<string, unknown>;
-                } catch {
-                    // Do nuffin
-                }
-            }
-            currentCache[this._id] = newVal;
-            writeFile(cacheFile, JSON.stringify(currentCache, null, 2));
-        });
     }
 
     /**
-     * Initializes this option by attempting to read its default value from the default file.
-     * If found, sets the current value.
+     * Set the given configuration value and write it to disk, if specified.
+     *
+     * @param value - The new value.
+     * @param writeDisk - Whether to write the changes to disk. Defaults to true.
      */
-    public createDefault(): void {
-        const rawData = readFile(defaultFile);
-
-        let defaultData: Record<string, unknown> = {};
-
-        if (rawData && rawData.trim() !== '') {
-            try {
-                defaultData = JSON.parse(rawData) as Record<string, unknown>;
-            } catch {
-                // do nuffin
-            }
+    public set(value: T, { writeDisk = true }: WriteDiskProps = {}): void {
```

Comment:
```suggestion
    public set = (value: T, { writeDisk = true }: WriteDiskProps = {}): void => {
```
Because Variable<T> is wrapped in a Proxy, its methods are treated as properties holding functions. Let's get typescript to stop whining by redefining `set` as an arrow function so it matches the base type.

Reply:
This seems to make typescript complain even more https://github.com/Jas-SinghFSU/HyprPanel/actions/runs/13882049556/job/38841835707?pr=684 :thinking: